legal-api/src/legal_api/models/filing.py

```diff
@@ -422,13 +422,14 @@ def locked(self):
 
         return False
 
-    def set_processed(self):
+    def set_processed(self, business_type):
         """Assign the completion and effective dates, unless they are already set."""
         if not self._completion_date:
             self._completion_date = datetime.utcnow()
-        if (self.effective_date is None or
-            (self.payment_completion_date
-             and self.effective_date < self.payment_completion_date)):  # pylint: disable=W0143; hybrid property
+        # In the case of an annual report or change of director filings on a COOP or BEN then the effective date can be before the payment date
+        if((self.filing_type == Filing.FILINGS['annualReport'] or self.filing_type == Filing.FILINGS['changeOfDirectors']) and (business_type in {'CP', 'BEN'})):
```

Comment:
For the reviewers, is there any better way to check the business_type? Because I don't want to create a circular import by importing the Business class